app/controllers/Emergency.scala

```diff
@@ -31,22 +42,98 @@ object Emergency extends Controller with PanDomainAuthActions {
       }
     }).getOrElse {
-      unauthorised("No existing login session found, unable to log you in.")
+      unauthorised("No existing login session found, unable to log you in.", reissueTopic)
+    }
+  }
+
+  def requestCookieLink = EmergencySwitchIsOnAction { req =>
+    Ok(views.html.emergency.newCookieIssued())
+  }
+
+  def sendCookieLink = EmergencySwitchIsOnAction { req =>
+
+    val tokenIssuedAt = DateTime.now().getMillis
+    val emailAddress= req.body.asFormUrlEncoded.get("email")(0)
+
+    if (emailAddress.matches("^[a-z]+\\.[a-z]+@(guardian.co.uk|theguardian.com)$")) {
+
+      val token = Random.alphanumeric.take(20).mkString
+
+      val cookieIssue = NewCookieIssue(token, emailAddress,
+        tokenIssuedAt, false)
+
+      try {
+        val userOpt = Scanamo.put[NewCookieIssue](AWS.dynamoDbClient)(loginConfig.tokensTableName)(cookieIssue)
+        val ses = new SES(loginConfig.sesClient, loginConfig)
+        ses.sendCookieEmail(token, emailAddress)
+
+        Ok(views.html.emergency.emailSent())
+      }
+      catch {
+        case e: Throwable => InternalServerError(e.toString())
+      }
     }
+    else {
+      BadRequest("Only guardian email addresses are supported")
+    }
+  }
+
+  def issueNewCookie(userToken: String) = EmergencySwitchIsOnAction { req =>
+
+    val newCookieTopic = "A new cookie has not been created"
+    val issueNewCookieTopic = "New cookie has not been created"
+    val tenMinutesInMilliSeconds = 600000
+
+    val tableName = loginConfig.tokensTableName
+    val tokenOpt: Option[Xor[DynamoReadError, NewCookieIssue]] = Scanamo.get[NewCookieIssue](AWS.dynamoDbClient)(tableName)('id -> s"$userToken")
+    tokenOpt.map {
+      case Left(error) => {
+        Logger.warn(s"Error when reading entry with $userToken from dynamo. A new cookie will not be issued")
+        unauthorised("Checking your access token failed. You will not be issued with a new ", issueNewCookieTopic)
+      }
+      case Right(tokenEntry: NewCookieIssue) => {
+        if (!tokenEntry.used) {
+          val tokenAgeInMilliseconds = DateTime.now().getMillis - tokenEntry.requested
+          if (tokenAgeInMilliseconds > tenMinutesInMilliSeconds) {
+            Unauthorized(views.html.emergency.newCookieFailure("Your link has expired. Could not create a new cookie"))
+          }
+          else {
+            val updatedTokenEntry = Scanamo.put[NewCookieIssue](AWS.dynamoDbClient)(tableName)(tokenEntry.copy(used=true))
+            val expires = (DateTime.now() + cookieLifetime).getMillis
+            val names = tokenEntry.email.split("\\.")
+            val firstName = names(0).capitalize
+            val lastName = names(1).split("@")(0).capitalize
+            val user = User(firstName, lastName, tokenEntry.email, None)
+            val newAuthUser = AuthenticatedUser(user, loginConfig.appName, Set(loginConfig.appName), expires, true)
+            val authCookies = generateCookies(newAuthUser)
+
```

Comment:
For readability it might be worth worth putting most of the `else` block (L111-L118) into a inner method called `issueNewCookie`.